kube-webhook-certgen/v1.4.0/rockcraft.yaml

```diff
@@ -0,0 +1,166 @@
+license: Apache-2.0
+
+# NOTE(aznashwan): the `kube-webhook-certgen` image is versioned independently
+# from the main `nginx-ingress-controller` image, with 'v1.4.0' being the tag
+# corresponding to the `controller-v1.10.1`
+# https://github.com/kubernetes/ingress-nginx/pull/11033
+# https://github.com/kubernetes/ingress-nginx/releases/tag/controller-v1.10.0
+version: v1.4.0
+
+# NOTE(aznashwan): upstream image is build from `gcr.io/distroless/static:nonroot`:
+# https://github.com/kubernetes/ingress-nginx/blob/controller-v1.10.1/images/kube-webhook-certgen/rootfs/Dockerfile#L27
+# The base `distroless/static:nonroot` image is built using Bazel and is basically
+# just a very minimal Debian-based image with a `nonroot` user added to it.
+# https://github.com/GoogleContainerTools/distroless/blob/main/base
+base: bare
+build-base: ubuntu@22.04
+platforms:
+  amd64:
+  arm64:
+
+services:
+  kube_webhook_certgen:
+    startup: enabled
+    override: replace
+
+    # NOTE(aznashwan): simple script which sleeps for 1 second before
+    # exec-ing the actual kube-webhook-certgen binary:
+    # https://github.com/canonical/pebble/issues/240#issuecomment-1599722443
+    command: busybox sh /pebble-entrypoint.sh [ create ]
+
+    on-success: shutdown
+    on-failure: shutdown
+
+    user: nonroot
+    group: nonroot
+    working-dir: /
+
+entrypoint-service: kube_webhook_certgen
+
+parts:
+
+  # NOTE(aznashwan): upstream image is build from `gcr.io/distroless/static:nonroot`:
+  # https://github.com/kubernetes/ingress-nginx/blob/controller-v1.10.1/images/kube-webhook-certgen/rootfs/Dockerfile#L27
+  # The base `distroless/static:nonroot` image is built using Bazel and is basically
+  # just a very minimal Debian-based image with a `nonroot` user added to it.
+  # https://github.com/GoogleContainerTools/distroless/blob/main/base
+  install-base-packages:
+    plugin: nil
+    stage-packages:
+      - base-files
+      - netbase
+      - tzdata
+      - passwd
+      - login
+      # https://github.com/GoogleContainerTools/distroless/blob/a019fc2/base/base.bzl#L118
+      - libssl3
+    override-build: |
+      set -eux -o pipefail
+
+      # Manually include nsswitch.conf as seen here:
+      # https://github.com/GoogleContainerTools/distroless/blob/a019fc2/base/base.bzl#L92
+      etc="$CRAFT_PART_INSTALL/etc"
+      mkdir -p $etc
+      cp $CRAFT_PROJECT_DIR/nsswitch.conf $etc/nsswitch.conf
+
+  include-busybox-debug:
+    after: ["install-base-packages"]
+    plugin: nil
+    build-packages:
+      - busybox-static
+
+    build-environment:
+      # NOTE(aznashwan): the upstream dstroless images also have variants with
+      # BusyBox included to facilitate debugging, which we will conditionally
+      # include based on this environment variable.
+      # https://github.com/GoogleContainerTools/distroless/blob/a019fc2/base/base.bzl#L157
+      # TODO(aznashwan): set this back to 0:
```

Comment:
set back to 0

Reply:
Unfortunately can't since we need busybox (or some other combination of a shell + a handful of individual coreutils) to [run the entrypoint](https://github.com/canonical/ingress-nginx-rocks/pull/4/files#diff-aaa9db11147a6be8c3165b328b48ddfbc5b391ddca4653dd90cfaa903fa57444R40).

This is due to the `kube-webhook-certgen` binary executing in less than 1 second, so Pebble does NOT heed the `on-success: shutdown` and just hangs indefinitely, even if the exit code is 0, hence [the need for the hacky entrypoint script too](https://github.com/canonical/ingress-nginx-rocks/pull/4/files#diff-c43f8987e7e605118e8c86a64c0a01002ddf98925e9638710b9304eae9b84435R10-R12) too...

I will, however, be removing this conditional and updating the comment.